tests/phpunit/api/v3/ContactTest.php

```diff
@@ -4333,6 +4359,35 @@ public function testContactGreetingsCreateWithCustomField() {
     $this->customGroupDelete($ids['custom_group_id']);
   }
 
+  /**
+   * Test that smarty variables are parsed if they exist in the greeting template.
+   *
+   * @param int $version
+   *
+   * @dataProvider versionThreeAndFour
+   * @throws \CRM_Core_Exception
+   */
+  public function testGreetingParseSmarty($version) {
```

Comment:
@eileenmcnaughton Can you tweak the comments a little bit so it is clear that we are testing a string with both CiviCRM tokens and smarty variables?

Reply:
@mattwire the test ONLY  has smarty  variables - ie the IF - the test is to make sure they DO get parsed

Reply:
`{contact.first_name}` will get parsed via CRM_Utils_Token / tokenProcessor whereas the `{if}` will be processed by smarty?

Reply:
Oh - right yeah I guess so - OK